core/src/test/java/hudson/model/UserTest.java

```diff
@@ -43,4 +43,31 @@ public void blankIdsOrFullNamesShouldNotBeAllowed() {
         assertThat("Blank user IDs should not be allowed", User.isIdOrFullnameAllowed("      "), is(false));
     }
     
+    @Test
+    @Issue("JENKINS-35967")
+    public void shoudNotAllowIllegalRestrictedNamesInWrongCase() {
+        assertIdOrFullNameNotAllowed("system");
+        assertIdOrFullNameNotAllowed("System");
+        assertIdOrFullNameNotAllowed("SYSTEM");
+        assertIdOrFullNameNotAllowed("syStem");
+        assertIdOrFullNameNotAllowed("sYstEm");
+    }
+    
+    @Test
+    @Issue("JENKINS-35967")
+    public void shoudNotAllowIllegalRestrictedNamesEvenIfTrimmed() {
+        for (String username : User.getIllegalPersistedUsernames()) {
+            assertIdOrFullNameNotAllowed(username);
+            assertIdOrFullNameNotAllowed(" " + username);
+            assertIdOrFullNameNotAllowed(username + " ");
+            assertIdOrFullNameNotAllowed("      " + username + "    ");
+            assertIdOrFullNameNotAllowed("\t" + username + "\t");  
```

Comment:
I intended for validation against the restricted names, not stripping accents before persistence.